Lib/test/test_shutil.py

```diff
@@ -2320,24 +2320,39 @@ def test_stty_match(self):
             del env['LINES']
             del env['COLUMNS']
             actual = shutil.get_terminal_size()
+            self.assertEqual(expected, actual)
 
-        self.assertEqual(expected, actual)
+            # Falls back to stdout.
+            with support.swap_attr(sys, '__stdin__', None), \
+                    support.swap_attr(sys, '__stderr__', None):
+                actual = shutil.get_terminal_size()
+                self.assertEqual(expected, actual)
+
+            # Falls back to stderr.
```

Comment:
The test is decorated with `@unittest.skipUnless(os.isatty(sys.__stdout__.fileno()), "not on tty")`. What if stderr is not a TTY?